content/articles/Integrating-YouTube-videos-into-a-Django-app/index.md

```diff
@@ -0,0 +1,309 @@
+
+### Prerequisites
+
+1.  Install and set up a programming environment for python 3
+2.  Have some fundamental knowledge in python and the Django framework
+
+#### Step 1: Setting up the environment and downloading required packages
+
+We will begin by creating a directory that will contain our project.
+Open the command line and type in the following:
+
+```bash
+mkdir projectX
+```
+
+Change your current directory to projectX:
+
+```bash
+cd projectX
+```
+
+Then create and activate the virtual environment respectively (for windows):
+
+```bash
+py -m venv .venv
+```
+
+```bash
+.venv\Scripts\activate.bat
+```
+
+Now we can install Django and Django-embed-video simultaneously.
+
+ - Django : this is the Django framework package.
+ - Django-embed-video : this is the Django app that makes the embeding of videos from Youtube, Vimeo and music from    soundcloud easy.
+
+```
+pip install django django-embed-video
+```
+
+Create a Django project:
+
+```bash
+django-admin startproject demo
+```
+
+Change directory to demo
+
+```bash
+cd demo
+```
+
+Create a Django app
+
+```bash
+py manage.py startapp blog
+```
+
+### Step 2: Setting up the backend
+
+#### 2.1: Register apps
+
+In `settings.py` under `INSTALLED_APPS` add our new apps embed_video and blog
+
+```python
+INSTALLED_APPS = [
+#...
+'embed_video',
+'blog',
+]
+```
+
+#### 2.2: Creating the models
+
+Our article will have three parts: the title, the body where the content is and lastly the YouTube video.
+
+In `models.py` create the following model and remember to make the necessary import `EmbedVideoField` as shown below.
+
+```python
+from  embed_video.fields  import  EmbedVideoField
+#Create your models here.
+class  tutorial(models.Model):
+	tutorial_Title = models.CharField(max_length=200)
+	tutorial_Body = models.TextField()
+	tutorial_Video = EmbedVideoField()
+
+	class  Meta:
+		verbose_name_plural = "Tutorial"
+
+	def  __str__(self):
+		return  str(self.tutorial_Title) if  self.tutorial_Title  else  " "
+```
+
+Then run the following commands respectively to make migrations and migrate our model.
+
+```bash
+py manage.py makemigrations
+```
+
+```bash
+py manage.py migrate
+```
+
+#### 2.3: Registering the models
+
+In order to have access to our model using Django admin we have to register our model in `admin.py`.
+Import `AdminVideoMixin` and `tutorial` then register your model as shown below.
+
+```python
+from  embed_video.admin  import  AdminVideoMixin
+from .models  import  tutorial
+#Register your models here.
+
+class  tutorialAdmin(AdminVideoMixin, admin.ModelAdmin):
+	pass
+
+admin.site.register(tutorial, tutorialAdmin)
+```
+
+#### 2.4: Creating the views
+
+We will need two views namely `blog` and `blog_detail`. The blog will be responsible for fetching all the objects when `blog.html` is requested.
+Blog_detail will be responsible for fetching the contents of a specific object when `blogdetail.html` is requested.
```

Comment:
This is an example of a simple paragraph. Try adding explanatory sentences like these to all of the other ones that are too short.